docs/whats-new/preview-4.md

```diff
@@ -0,0 +1,647 @@
+
+var blobs = builder.AddAzureStorage("storage")
+                   .RunAsEmulator()
+                   .AddBlobs("blobs");
+
+builder.AddProject<Projects.Api>("api")
+       .WithReference(blobs);
+
+builder.Build().Run();
+```
+
+Each `RunAsEmulator` method has a callback that enables customization of the emulator container resource. For example, you can change the image, tag, or add additional volumes.
+
+**Azure storage emulator (Azurite)**
+
+```csharp
+var storage = builder.AddAzureStorage("storage").RunAsEmulator(container =>
+{
+    container.UsePersistence();
+});
+```
+
+**CosmosDB emulator**
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var db = builder.AddAzureCosmosDB("cosmos")
+                .RunAsEmulator()
+                .AddDatabase("db");
+
+builder.AddProject<Projects.Api>("api")
+       .WithReference(db);
+
+builder.Build().Run();
+```
+
+## The deployment manifest
+
+The aspire manifest has undergone a significant overhaul to support the new primitives and introduced in preview 4. The new manifest focuses on a few key primitives to enable a wide range of scenarios:
+
+| Resource type   | Description                                                   |
+|-----------------|---------------------------------------------------------------|
+| `project.v0`    | .NET project files                                            |
+| `container.v0`  | Container images                                              |
+| `dockerfile.v0` | Docker files                                                  |
+| `parameter.v0`  | External Parameters                                           |
+| `value.v0`      | References to other resources (or a combination of resources) |
+
+### Azure specific resources
+
+| Resource type    | Description           |
+|------------------|-----------------------|
+| `azure.bicep.v0` | Azure Bicep templates |
+
+Tool authors can support this very small set of resource types to model lots of different apps!
+
+We've deprecated the abstract resource types that were supported in previous versions. Deployment tools may still support those resource types, but they are no longer part of the core manifest schema.
+
+### Visual Studio Publish to Azure
+
+This release we've enabled a Visual Studio publish experience for Aspire applications.
+
+TODO: Deployment to Azure via Visual Studio support (@bradygaster @timheuer)
+
+## Azure Developer CLI
+
+The Azure Developer CLI (azd) has been updated to support the new manifest types introduced in preview 4. Azd supports prompting for parameter resources, and supports the bicep resource which enables deploying any thing can be described using a bicep module.
+
+In addition to new features, azd will automatically create secrets in Azure Container Apps for any parameter marked as a secret, and any environment variable that references a secret or uses a connection string.
+
+TODO: Expand with images
```

Comment:
@rajeshkamal5050 we need some more content here. I can add some more but @vhvb1989 would know best. I think beyond the new resource types I would like to see more details around parameters as that's a new key experience we're delivering.